inductiva/fluids/scenarios/wind_terrain/wind_terrain.py

```diff
@@ -0,0 +1,205 @@
+                                              COMMANDS_TEMPLATE_FILE_NAME)
+
+        with tempfile.NamedTemporaryFile() as commands_file:
+            inductiva.utils.templates.replace_params_in_template(
+                template_path=commands_template_path,
+                params={"n_cores": self.n_cores},
+                output_file_path=commands_file.name,
+            )
+
+            commands = self.read_commands_from_file(commands_file.name)
+
+        return commands
+
+    @singledispatchmethod
+    def create_input_files(self, simulator: inductiva.simulation.Simulator):
+        pass
+
+
+@WindTerrain.create_input_files.register
+def _(self, simulator: fluids.simulators.OpenFOAM, input_dir):  # pylint: disable=unused-argument
+    """Creates OpenFOAM simulation input files."""
+
+    # The WindTunnel with OpenFOAM requires changing multiple files
+    template_files_dir = os.path.join(SCENARIO_TEMPLATE_DIR,
+                                      OPENFOAM_TEMPLATE_INPUT_DIR, FILES_SUBDIR)
+
+    # Copy all files from the template dir to the input directory
+    shutil.copytree(template_files_dir,
+                    input_dir,
+                    dirs_exist_ok=True,
+                    symlinks=True)
+
+    inductiva.utils.templates.batch_replace_params_in_template(
+        templates_dir=input_dir,
+        template_filenames=[
+            os.path.join("system", "blockMeshDict_template.openfoam.jinja"),
+            os.path.join("system", "controlDict_template.openfoam.jinja"),
+            os.path.join("system", "decomposeParDict_template.openfoam.jinja"),
+            os.path.join("system", "snappyHexMeshDict_template.openfoam.jinja"),
+            os.path.join("system", "topoSetDict_template.openfoam.jinja"),
+            os.path.join("0", "include",
+                         "ABLConditions_template.openfoam.jinja"),
+            os.path.join("constant", "fvOptions_template.openfoam.jinja")
+        ],
+        params={
+            "terrain_bounds": self.terrain.bounds,
+            "terrain_center": self.terrain.center,
+            "terrain_path": TERRAIN_FILENAME,
+            "wind_flow_dir": self.wind_direction,
+            "flow_position": self.wind_position,
+            "top_boundary_height": self.top_boundary_height,
+            "num_iterations": self.num_iterations,
+            "n_cores": self.n_cores,
+        },
+        output_filename_paths=[
+            os.path.join(input_dir, "system", "blockMeshDict"),
+            os.path.join(input_dir, "system", "controlDict"),
+            os.path.join(input_dir, "system", "decomposeParDict"),
+            os.path.join(input_dir, "system", "snappyHexMeshDict"),
+            os.path.join(input_dir, "system", "topoSetDict"),
+            os.path.join(input_dir, "0", "include", "ABLConditions"),
+            os.path.join(input_dir, "constant", "fvOptions")
+        ],
+        remove_templates=True,
+    )
```

Comment:
Do you prefer this syntax over having a single file with initial/boundary conditions that other files read from?

Reply:
I know this is over complicated for now, but I prefer it to tweak things more easily and know where they are. When the scenario is more established, I will move to the idea done in the HeatSink.